File: modules/wem-api/src/main/java/com/enonic/wem/query/ast/FieldOrderExpr.java
package com.enonic.wem.query.ast;

public final class FieldOrderExpr
    extends OrderExpr
{
    private final FieldExpr field;

    public FieldOrderExpr( final FieldExpr field, final Direction direction )
    {
        super( direction );
        this.field = field;
    }

    public FieldExpr getField()
    {
        return this.field;
    }

    @Override
    public String toString()
    {
        return this.field + " " + getDirection();
    }
}

File: modules/admin/admin-ui/src/main/resources/web/admin/common/js/content/FieldOrderExpr.ts
module api.content {

    export class FieldOrderExpr extends OrderExpr {

        private fieldName: string;

        constructor(builder: FieldOrderExprBuilder) {
            super(builder);
            this.fieldName = builder.fieldName;
        }

        getFieldName(): string {
            return this.fieldName;
        }

        toJson(): json.OrderExprJson {
            return {
                "fieldName": this.fieldName,
                "direction": this.getDirection()
            };
        }

        toString() {
            return this.fieldName + " " + super.getDirection();
        }

        equals(o: api.Equitable): boolean {
            if (!super.equals(o)) {
                return false;
            }
            if (!api.ObjectHelper.iFrameSafeInstanceOf(o, FieldOrderExpr)) {
                return false;
            }
            var other = <FieldOrderExpr>o;
            if (this.fieldName.toLowerCase() != other.getFieldName().toLowerCase()) {
                return false;
            }
            return true;
        }
    }

    export class FieldOrderExprBuilder extends OrderExprBuilder {

        fieldName: string;

        constructor(json?: json.OrderExprJson) {
            super(json);
            if (json) {
                this.fieldName = json.fieldName;
            }
        }

        public setFieldName(value: string): FieldOrderExprBuilder {
            this.fieldName = value;
            return this;
        }

        public build(): FieldOrderExpr {
            return new FieldOrderExpr(this);
        }
    }
}